### PR TITLE
Linux port: vulkan layer init & intergration

### PR DESCRIPTION
**Pull request by ronthecookie**
This PR adds @xytovl's [vulkan-wsi-layer](https://github.com/xytovl/vulkan-wsi-layer) to the main ALVR tree and integrates it with the `session.json` / `Settings` class to allow for dynamic resolution and refresh rate and also with a connection to transfer Vulkan images to the ALVR `CEncoder`.

~~I commented the logging bits in `Settings.cpp` out for now so it builds but I suspect you'll want them back soon.~~
Added logging stuff back with my own `Logger.cpp` implementation.

Relevant [wiki page here](https://github.com/alvr-org/ALVR/wiki/Linux-Support-development-progress#2021-04-01).

### CODE REVIEW

File: alvr/server/cpp/tools/vulkan-layer/DCO.txt [hunk 1]
```diff
@@ -0,0 +1,37 @@
+Developer Certificate of Origin
```
Comment (xytovl) [inline alvr/server/cpp/tools/vulkan-layer/DCO.txt:1]: Do we have to keep this file? the license itself should be sufficient.

File: alvr/server/cpp/tools/vulkan-layer/layer/alvr_x86_64.json [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+		"implementation_version": "1",
+		"description": "ALVR display intercept layer",
+		"enable_environment": {
+			"SteamAppId":"250820"
+		},
+		"functions": {
+			"vkGetInstanceProcAddr": "wsi_layer_vkGetInstanceProcAddr",
+			"vkGetDeviceProcAddr": "wsi_layer_vkGetDeviceProcAddr"
+		},
+		"instance_extensions": [
```
Comment (xytovl) [inline alvr/server/cpp/tools/vulkan-layer/layer/alvr_x86_64.json:17]: we should list the 2 more extensions provided by the layer
VK_EXT_acquire_xlib_display
VK_KHR_display

**Review by xytovl — COMMENTED** [review]
A few minor suggestions.

Do we want to rename all `wsi_` prefixes to `alvr_`? I don't think there can be collisions with the proper wsi layer, but we are not doing any window system integration anymore.

**Comment by ronthecookie** [comment]
> Do we want to rename all `wsi_` prefixes to `alvr_`? I don't think there can be collisions with the proper wsi layer, but we are not doing any window system integration anymore.

@xytovl That seems like unnecessary work to me but if you want to do it, go for it.

**Comment by ronthecookie** [comment]
Proceeded by https://github.com/alvr-org/ALVR/pull/604.

**Pull request closed without merging**